transport-classes-io_uring/src/main/java/io/netty/channel/uring/IoUringBufferRing.java

```diff
@@ -57,12 +51,22 @@ final class IoUringBufferRing {
         this.source = ioUringIoHandler;
         this.byteBufAllocator = byteBufAllocator;
         this.exhaustedEvent = new IoUringBufferRingExhaustedEvent(bufferGroupId);
+        fill();
     }
 
-    void refillIfNecessary() {
-        if (!hasSpareBuffer) {
-            refill();
+    private boolean fill() {
+        boolean fillSomething = false;
+        for (short i = 0; i < entries; i++) {
+            if (buffers[i] == null) {
+                addBuffer(i);
```

Comment:
We can first add all buffer to the ring and then advance the tail. 
This way, the number of `putShortOrdered` operations can be reduced.

Reply:
Let's keep it simple for now. Might do this as a followup